### ISSUE TITLE
Test cluster upgrading on Equinix Metal - Test Release KubeOne 1.5

### ISSUE THREAD

**Issue by xmudrii** [kubermatic/kubeone#2251]
This is a ticket for tracking the progress of testing cluster upgrading on Equinix Metal.

Follow instructions in #2247. Equinix Metal **has** automated E2E tests.

### From v1.23.9 To v1.24.3

* Containerd + External CCM
  * [x] Ubuntu: `/test pull-kubeone-e2e-equinixmetal-default-upgrade-containerd-external-from-v1.23.9-to-v1.24.3`
  * [x] CentOS: `/test pull-kubeone-e2e-equinixmetal-centos-upgrade-containerd-external-from-v1.23.9-to-v1.24.3`
  * [x] Flatcar: `/test pull-kubeone-e2e-equinixmetal-flatcar-upgrade-containerd-external-from-v1.23.9-to-v1.24.3`

### From v1.22.12 To v1.23.9

* Containerd + External CCM
  * [x] Ubuntu: `/test pull-kubeone-e2e-equinixmetal-default-upgrade-containerd-external-from-v1.22.12-to-v1.23.9`
  * [x] CentOS: `/test pull-kubeone-e2e-equinixmetal-centos-upgrade-containerd-external-from-v1.22.12-to-v1.23.9`
  * [x] Flatcar: `/test pull-kubeone-e2e-equinixmetal-flatcar-upgrade-containerd-external-from-v1.22.12-to-v1.23.9`
* Docker + External CCM
  * [x] Ubuntu: `/test pull-kubeone-e2e-equinixmetal-default-upgrade-docker-external-from-v1.22.12-to-v1.23.9`
  * [x] CentOS: `/test pull-kubeone-e2e-equinixmetal-centos-upgrade-docker-external-from-v1.22.12-to-v1.23.9`
  * [x] Flatcar: `/test pull-kubeone-e2e-equinixmetal-flatcar-upgrade-docker-external-from-v1.22.12-to-v1.23.9`

**Comment by embik** [comment]
Even on the latest kubeone `master` build, I had to manually delete the machine-controller-webhook Pod when doing the 1.22.12 -> 1.23.9 upgrade, because the webhook could not talk to the Equinix API:

```
error: machinedeployments.cluster.k8s.io "marvinqa-pool1" could not be patched: admission webhook "machinedeployments.machine-controller.kubermatic.io" denied the request: validation failed: failed to list facilities: Get "https://api.equinix.com/metal/v1/facilities": dial tcp: lookup api.equinix.com: i/o timeout
```

@xmudrii FYI.

**Comment by xmudrii** [comment]
@embik This is sort of expected because KubeOne 1.4.7 doesn't have a fix needed for Flatcar clusters (#2301). That said, I'll go ahead and close this issue.
